test/lazy/_lazy_tensor_test_case.py

```diff
@@ -143,6 +143,26 @@ def test_add_diag(self):
         actual = evaluated + other_diag.diag()
         self.assertTrue(approx_equal(res, actual))
 
+    def test_root_decomposition(self):
```

Comment:
Hmm before looking at this I guess I wasn't really aware that RootLazyTensor matmul is actually multiplying the LazyTensor with the rhs, not the root. I feel this is slightly confusing but I guess reasonable....